File: backend/models/diagnosis.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, Text, JSON, Float, Enum
from sqlalchemy.sql import func
from database import Base
import enum

class DiagnosisType(enum.Enum):
    BRIGHTNESS = "brightness"  # 亮度检测
    BLUE_SCREEN = "blue_screen"  # 蓝屏检查
    CLARITY = "clarity"  # 清晰度检查
    SHAKE = "shake"  # 抖动检查
    FREEZE = "freeze"  # 冻结检测
    COLOR_CAST = "color_cast"  # 偏色检测
    OCCLUSION = "occlusion"  # 遮挡检测
    NOISE = "noise"  # 噪声检测
    CONTRAST = "contrast"  # 对比度检测
    MOSAIC = "mosaic"  # 马赛克检测
    FLOWER_SCREEN = "flower_screen"  # 花屏检测
    SIGNAL_LOSS = "signal_loss"  # 信号丢失
    LENS_DIRTY = "lens_dirty"  # 镜头脏污
    FOCUS_BLUR = "focus_blur"  # 焦点模糊

class DiagnosisStatus(enum.Enum):
    NORMAL = "normal"  # 正常
    WARNING = "warning"  # 警告
    ERROR = "error"  # 错误
    CRITICAL = "critical"  # 严重

class TaskStatus(enum.Enum):
    PENDING = "PENDING"  # 待执行
    RUNNING = "RUNNING"  # 运行中
    COMPLETED = "COMPLETED"  # 已完成
    FAILED = "FAILED"  # 失败
    CANCELLED = "CANCELLED"  # 已取消

class TaskType(enum.Enum):
    DIAGNOSIS = "DIAGNOSIS"  # 诊断任务
    AI_DETECTION = "AI_DETECTION"  # AI检测任务
    MONITORING = "MONITORING"  # 监控任务
    MAINTENANCE = "MAINTENANCE"  # 维护任务

class DiagnosisTask(Base):
    __tablename__ = "diagnosis_tasks"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False, comment="任务名称")
    description = Column(Text, comment="任务描述")
    
    # 任务类型
    task_type = Column(Enum(TaskType), default=TaskType.DIAGNOSIS, comment="任务类型")
    
    # 模板关联
    template_id = Column(Integer, comment="诊断模板ID")
    
    # 诊断配置 - 匹配实际数据库结构
    camera_ids = Column(JSON, default=list, comment="摄像头ID列表")
    camera_groups = Column(JSON, default=list, comment="摄像头组列表")
    diagnosis_types = Column(JSON, default=list, comment="诊断类型列表")
    diagnosis_config = Column(JSON, default=dict, comment="诊断配置参数")
    
    # 调度配置 - 匹配实际数据库结构
    schedule_type = Column(String(50), comment="调度类型")
    schedule_config = Column(JSON, default=dict, comment="调度配置")
    cron_expression = Column(String(100), comment="Cron表达式")
    interval_minutes = Column(Integer, comment="间隔分钟数")
    threshold_config = Column(JSON, default=dict, comment="阈值配置")
    
    # 状态信息
    status = Column(Enum(TaskStatus), default=TaskStatus.PENDING, comment="任务状态")
    is_active = Column(Boolean, default=True, comment="是否启用")
    enabled = Column(Boolean, default=True, comment="是否启用")
    assigned_worker = Column(String(100), comment="分配的worker节点ID")
    
    # 执行信息 - 匹配实际数据库结构
    started_at = Column(DateTime(timezone=True), comment="任务开始时间")
    last_run_time = Column(DateTime(timezone=True), comment="最后执行时间")
    next_run_time = Column(DateTime(timezone=True), comment="下次执行时间")
    total_runs = Column(Integer, default=0, comment="总执行次数")
    success_runs = Column(Integer, default=0, comment="成功执行次数")
    
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), comment="更新时间")
    created_by = Column(String(50), comment="创建人ID")

class AlarmRule(Base):
    """告警规则模型"""
    __tablename__ = "alarm_rules"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False, comment="规则名称")
    description = Column(Text, comment="规则描述")
    
    # 规则配置
    diagnosis_types = Column(JSON, default=list, comment="适用的诊断类型")
    camera_ids = Column(JSON, default=list, comment="适用的摄像头ID列表")
    camera_groups = Column(JSON, default=list, comment="适用的摄像头组")
    
    # 触发条件
    severity_level = Column(String(20), comment="触发的严重程度级别")
    threshold_config = Column(JSON, default=dict, comment="阈值配置")
    frequency_limit = Column(Integer, default=0, comment="频率限制(分钟内最多触发次数)")
    
    # 通知配置
    notification_channels = Column(JSON, default=list, comment="通知渠道")
    notification_template = Column(Text, comment="通知模板")
    
    # 状态
    is_enabled = Column(Boolean, default=True, comment="是否启用")
    priority = Column(Integer, default=1, comment="优先级")
    
    # 统计信息
    trigger_count = Column(Integer, default=0, comment="触发次数")
    last_triggered_at = Column(DateTime(timezone=True), comment="最后触发时间")
    
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), comment="更新时间")
    created_by = Column(String(50), comment="创建人ID")

class NotificationChannel(Base):
    """通知渠道模型"""
    __tablename__ = "notification_channels"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False, comment="渠道名称")
    type = Column(String(50), nullable=False, comment="渠道类型: email, sms, webhook, dingtalk, wechat")
    description = Column(Text, comment="渠道描述")
    
    # 配置信息
    config = Column(JSON, default=dict, comment="渠道配置")
    
    # 状态
    is_enabled = Column(Boolean, default=True, comment="是否启用")
    
    # 统计信息
    send_count = Column(Integer, default=0, comment="发送次数")
    success_count = Column(Integer, default=0, comment="成功次数")
    last_used_at = Column(DateTime(timezone=True), comment="最后使用时间")
    
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), comment="更新时间")
    created_by = Column(String(50), comment="创建人ID")

class NotificationLog(Base):
    """通知日志模型"""
    __tablename__ = "notification_logs"
    
    id = Column(Integer, primary_key=True, index=True)
    alarm_id = Column(Integer, comment="告警ID")
    rule_id = Column(Integer, comment="规则ID")
    channel_id = Column(Integer, comment="通知渠道ID")
    
    # 通知信息
    title = Column(String(200), comment="通知标题")
    content = Column(Text, comment="通知内容")
    recipients = Column(JSON, default=list, comment="接收人列表")
    
    # 发送状态
    status = Column(String(20), default="pending", comment="发送状态: pending, sent, failed")
    error_message = Column(Text, comment="错误信息")
    retry_count = Column(Integer, default=0, comment="重试次数")
    
    # 时间信息
    sent_at = Column(DateTime(timezone=True), comment="发送时间")
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")
    created_by = Column(String(50), comment="创建人ID")
    
    def __repr__(self):
        return f"<DiagnosisTask(id={self.id}, name='{self.name}', status='{self.status}')>"

class DiagnosisResult(Base):
    __tablename__ = "diagnosis_results"
    
    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(Integer, nullable=False, comment="诊断任务ID")
    camera_id = Column(Integer, comment="摄像头ID")
    camera_name = Column(String(100), comment="摄像头名称")
    diagnosis_type = Column(String(50), comment="诊断类型")
    
    # 诊断状态 - 匹配数据库字段名
    diagnosis_status = Column(Enum(DiagnosisStatus), nullable=False, comment="诊断状态")
    score = Column(Float, comment="诊断分数")
    threshold = Column(Float, comment="阈值")
    is_abnormal = Column(Boolean, default=False, comment="是否异常")
    
    # 图像信息
    image_url = Column(String(500), comment="图像URL")
    thumbnail_url = Column(String(500), comment="缩略图URL")
    image_timestamp = Column(DateTime(timezone=True), comment="图像时间戳")
    
    # 处理信息
    processing_time = Column(Float, comment="处理时间(ms)")
    error_message = Column(Text, comment="错误信息")
    suggestions = Column(JSON, default=list, comment="建议")
    metrics = Column(JSON, default=dict, comment="指标数据")
    
    # 检测结果
    result_data = Column(JSON, default=dict, comment="详细结果数据")
    
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")
    
    def __repr__(self):
        return f"<DiagnosisResult(id={self.id}, camera_id={self.camera_id}, type='{self.diagnosis_type}', status='{self.diagnosis_status}')>"

class DiagnosisAlarm(Base):
    __tablename__ = "diagnosis_alarms"
    
    id = Column(Integer, primary_key=True, index=True)
    result_id = Column(Integer, nullable=False, comment="诊断结果ID")
    
    # 告警信息
    alarm_type = Column(String(50), nullable=False, comment="告警类型")
    severity = Column(String(20), default="warning", comment="严重程度")
    title = Column(String(200), nullable=False, comment="告警标题")
    description = Column(Text, comment="告警描述")
    
    # 阈值配置
    threshold_config = Column(JSON, default=dict, comment="阈值配置")
    current_value = Column(Float, comment="当前值")
    threshold_value = Column(Float, comment="阈值")
    
    # 确认状态
    is_acknowledged = Column(Boolean, default=False, comment="是否已确认")
    acknowledged_by = Column(Integer, comment="确认人ID")
    acknowledged_at = Column(DateTime(timezone=True), comment="确认时间")
    
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")

class DiagnosisTemplate(Base):
    __tablename__ = "diagnosis_templates"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False, comment="模板名称")
    description = Column(Text, comment="模板描述")
    
    # 模板配置 - 匹配实际数据库结构
    diagnosis_types = Column(JSON, default=list, comment="诊断类型列表")
    default_config = Column(JSON, default=dict, comment="默认配置")
    default_schedule = Column(JSON, default=dict, comment="默认调度配置")
    threshold_config = Column(JSON, default=dict, comment="阈值配置")
    
    # 状态 - 匹配实际数据库结构
    is_active = Column(Boolean, default=True, comment="是否启用")
    is_system = Column(Boolean, default=False, comment="是否系统模板")
    
    # 使用统计
    usage_count = Column(Integer, default=0, comment="使用次数")
    
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), comment="更新时间")
    created_by = Column(String(50), comment="创建人ID")

class DiagnosisStatistics(Base):
    __tablename__ = "diagnosis_statistics"
    
    id = Column(Integer, primary_key=True, index=True)
    date = Column(DateTime(timezone=True), nullable=False, comment="统计日期")
    
    # 统计维度
    camera_id = Column(Integer, comment="摄像头ID")
    diagnosis_type = Column(String(50), comment="诊断类型")
    
    # 统计数据
    total_checks = Column(Integer, default=0, comment="总检查次数")
    normal_count = Column(Integer, default=0, comment="正常次数")
    warning_count = Column(Integer, default=0, comment="警告次数")
    error_count = Column(Integer, default=0, comment="错误次数")
    critical_count = Column(Integer, default=0, comment="严重次数")
    
    # 性能指标
    avg_score = Column(Float, comment="平均分数")
    min_score = Column(Float, comment="最低分数")
    max_score = Column(Float, comment="最高分数")
    avg_processing_time = Column(Float, comment="平均处理时间")
    
    created_at = Column(DateTime(timezone=True), server_default=func.now(), comment="创建时间")
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), comment="更新时间")

File: web/src/api/diagnosis.ts
import request from '@/utils/request'
import type { ApiResponse } from '@/types/api'

// 诊断类型枚举
export enum DiagnosisType {
  BRIGHTNESS = 'brightness',
  BLUE_SCREEN = 'blue_screen',
  CLARITY = 'clarity',
  SHAKE = 'shake',
  FREEZE = 'freeze',
  COLOR_CAST = 'color_cast',
  OCCLUSION = 'occlusion',
  NOISE = 'noise',
  CONTRAST = 'contrast',
  MOSAIC = 'mosaic',
  FLOWER_SCREEN = 'flower_screen',
  SIGNAL_LOSS = 'signal_loss',
  LENS_DIRTY = 'lens_dirty',
  FOCUS_BLUR = 'focus_blur'
}

// 诊断状态枚举
export enum DiagnosisStatus {
  NORMAL = 'normal',
  WARNING = 'warning',
  ERROR = 'error',
  CRITICAL = 'critical'
}

// 任务状态枚举
export enum TaskStatus {
  PENDING = 'pending',
  RUNNING = 'running',
  COMPLETED = 'completed',
  FAILED = 'failed',
  CANCELLED = 'cancelled'
}

// 诊断任务接口
export interface DiagnosisTask {
  id: number
  name: string
  diagnosis_type: DiagnosisType
  target_id: number
  target_type: string
  template_id?: number
  template_name?: string
  config: Record<string, any>
  schedule_config: Record<string, any>
  threshold_config: Record<string, any>
  status: TaskStatus
  is_scheduled: boolean
  is_active: boolean
  last_run?: string
  next_run?: string
  run_count: number
  success_count: number
  error_count: number
  created_by: number
  created_by_name: string
  created_at: string
  updated_at: string
  description?: string
}

export interface DiagnosisTaskCreate {
  name: string
  diagnosis_type: DiagnosisType
  target_id: number
  target_type: string
  template_id?: number
  config?: Record<string, any>
  schedule_config?: Record<string, any>
  threshold_config?: Record<string, any>
  is_scheduled?: boolean
  description?: string
}

export interface DiagnosisTaskUpdate {
  name?: string
  diagnosis_type?: DiagnosisType
  target_id?: number
  template_id?: number
  config?: Record<string, any>
  schedule_config?: Record<string, any>
  threshold_config?: Record<string, any>
  is_scheduled?: boolean
  is_active?: boolean
  description?: string
}

// 诊断结果接口
export interface DiagnosisResult {
  id: number
  task_id: number
  task_name: string
  camera_name?: string  // 摄像头名称
  diagnosis_type?: string
  status: DiagnosisStatus
  result_data: Record<string, any>
  score?: number
  score_level?: string  // 分数等级评估
  score_description?: string  // 分数描述
  threshold?: number  // 阈值
  issues_found: Array<Record<string, any>>
  recommendations: string[]
  execution_time?: number
  processing_time?: number  // 处理时间(ms)
  error_message?: string
  image_url?: string
  thumbnail_url?: string
  created_at: string  // 检测时间
  detected_at?: string  // 检测时间别名
}

// 诊断告警接口
export interface DiagnosisAlarm {
  id: number
  result_id: number
  task_name: string
  alarm_type: string
  severity: string
  title: string
  description: string
  threshold_config: Record<string, any>
  current_value?: number
  threshold_value?: number
  is_acknowledged: boolean
  acknowledged_by?: number
  acknowledged_by_name?: string
  acknowledged_at?: string
  created_at: string
}

// 诊断模板接口
export interface DiagnosisTemplate {
  id: number
  name: string
  diagnosis_types: string[]
  default_config: Record<string, any>
  default_schedule: Record<string, any>
  threshold_config: Record<string, any>
  is_active: boolean
  is_system: boolean
  usage_count: number
  created_by: string
  created_by_name: string
  created_at: string
  updated_at: string
  description?: string
}

export interface DiagnosisTemplateCreate {
  name: string
  diagnosis_type: DiagnosisType
  config_template: Record<string, any>
  default_schedule?: Record<string, any>
  threshold_config?: Record<string, any>
  description?: string
  is_public?: boolean
}

// 诊断统计接口
export interface DiagnosisStats {
  total_tasks: number
  active_tasks: number
  running_tasks: number
  scheduled_tasks: number
  total_results: number
  success_results: number
  failed_results: number
  warning_results: number
  total_alarms: number
  unacknowledged_alarms: number
  critical_alarms: number
  avg_score: number
  by_type: Record<string, number>
  by_status: Record<string, number>
  trend_data: Array<Record<string, any>>
}

// 诊断任务API
export const diagnosisTaskApi = {
  // 获取任务列表
  getTasks: (params?: {
    page?: number
    page_size?: number
    search?: string
    diagnosis_type?: DiagnosisType
    status?: TaskStatus
    is_active?: boolean
    is_scheduled?: boolean
    target_type?: string
  }): Promise<ApiResponse<DiagnosisTask[]>> => {
    return request.get('/api/v1/diagnosis/tasks/', { params })
  },

  // 获取任务详情
  getTask: (id: number): Promise<ApiResponse<DiagnosisTask>> => {
    return request.get(`/api/v1/diagnosis/tasks/${id}`)
  },

  // 创建任务
  createTask: (data: DiagnosisTaskCreate): Promise<ApiResponse<DiagnosisTask>> => {
    return request.post('/api/v1/diagnosis/tasks/', data)
  },

  // 更新任务
  updateTask: (id: number, data: DiagnosisTaskUpdate): Promise<ApiResponse<DiagnosisTask>> => {
    return request.put(`/api/v1/diagnosis/tasks/${id}`, data)
  },

  // 删除任务
  deleteTask: (id: number): Promise<ApiResponse<void>> => {
    return request.delete(`/api/v1/diagnosis/tasks/${id}`)
  },

  // 执行任务
  runTask: (id: number): Promise<ApiResponse<void>> => {
    return request.post(`/api/v1/diagnosis/tasks/${id}/run`)
  },

  // 启用/禁用任务
  toggleTask: (id: number, is_active: boolean): Promise<ApiResponse<DiagnosisTask>> => {
    return request.put(`/api/v1/diagnosis/tasks/${id}`, { is_active })
  }
}

// 诊断结果分页响应接口
export interface DiagnosisResultListResponse {
  results: DiagnosisResult[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

// 诊断结果API
export const diagnosisResultApi = {
  // 获取结果列表
  getResults: (params?: {
    page?: number
    page_size?: number
    task_id?: number
    status?: DiagnosisStatus
    start_date?: string
    end_date?: string
  }): Promise<ApiResponse<DiagnosisResultListResponse>> => {
    return request.get('/api/v1/diagnosis/results/', { params })
  },

  // 获取结果详情
  getResult: (id: number): Promise<ApiResponse<DiagnosisResult>> => {
    return request.get(`/api/v1/diagnosis/results/${id}`)
  }
}

// 诊断告警API
export const diagnosisAlarmApi = {
  // 获取告警列表
  getAlarms: (params?: {
    page?: number
    page_size?: number
    alarm_type?: string
    severity?: string
    is_acknowledged?: boolean
    start_date?: string
    end_date?: string
  }): Promise<ApiResponse<DiagnosisAlarm[]>> => {
    return request.get('/api/v1/diagnosis/alarms/', { params })
  },

  // 确认告警
  acknowledgeAlarm: (id: number): Promise<ApiResponse<void>> => {
    return request.post(`/api/v1/diagnosis/alarms/${id}/acknowledge`)
  },

  // 批量确认告警
  batchAcknowledge: (ids: number[]): Promise<ApiResponse<void>> => {
    return request.post('/api/v1/diagnosis/alarms/batch-acknowledge', { alarm_ids: ids })
  }
}

// 诊断模板API
export const diagnosisTemplateApi = {
  // 获取模板列表
  getTemplates: (params?: {
    page?: number
    page_size?: number
    search?: string
    diagnosis_type?: DiagnosisType
    is_active?: boolean
    is_public?: boolean
  }): Promise<ApiResponse<DiagnosisTemplate[]>> => {
    return request.get('/api/v1/diagnosis/templates/', { params })
  },

  // 获取模板详情
  getTemplate: (id: number): Promise<ApiResponse<DiagnosisTemplate>> => {
    return request.get(`/api/v1/diagnosis/templates/${id}`)
  },

  // 创建模板
  createTemplate: (data: DiagnosisTemplateCreate): Promise<ApiResponse<DiagnosisTemplate>> => {
    return request.post('/api/v1/diagnosis/templates/', data)
  },

  // 更新模板
  updateTemplate: (id: number, data: Partial<DiagnosisTemplateCreate>): Promise<ApiResponse<DiagnosisTemplate>> => {
    return request.put(`/api/v1/diagnosis/templates/${id}`, data)
  },

  // 删除模板
  deleteTemplate: (id: number): Promise<ApiResponse<void>> => {
    return request.delete(`/api/v1/diagnosis/templates/${id}`)
  },

  // 启用/禁用模板
  toggleTemplate: (id: number, is_active: boolean): Promise<ApiResponse<DiagnosisTemplate>> => {
    return request.put(`/api/v1/diagnosis/templates/${id}`, { is_active })
  }
}

// 告警规则接口
export interface AlarmRule {
  id: number
  name: string
  description: string
  diagnosis_type: DiagnosisType
  camera_id?: number
  severity: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'
  threshold_config: Record<string, any>
  notification_channels: number[]
  is_enabled: boolean
  created_at: string
  updated_at: string
}

export interface AlarmRuleCreate {
  name: string
  description: string
  diagnosis_type: DiagnosisType
  camera_id?: number
  severity: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'
  threshold_config: Record<string, any>
  notification_channels: number[]
  is_enabled?: boolean
}

export interface AlarmRuleUpdate {
  name?: string
  description?: string
  diagnosis_type?: DiagnosisType
  camera_id?: number
  severity?: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'
  threshold_config?: Record<string, any>
  notification_channels?: number[]
  is_enabled?: boolean
}

// 通知渠道接口
export interface NotificationChannel {
  id: number
  name: string
  type: 'email' | 'sms' | 'webhook' | 'dingtalk' | 'wechat'
  config: Record<string, any>
  is_enabled: boolean
  created_at: string
  updated_at: string
}

export interface NotificationChannelCreate {
  name: string
  type: 'email' | 'sms' | 'webhook' | 'dingtalk' | 'wechat'
  config: Record<string, any>
  is_enabled?: boolean
}

export interface NotificationChannelUpdate {
  name?: string
  type?: 'email' | 'sms' | 'webhook' | 'dingtalk' | 'wechat'
  config?: Record<string, any>
  is_enabled?: boolean
}

// 通知日志接口
export interface NotificationLog {
  id: number
  alarm_id: number
  rule_id: number
  channel_id: number
  title: string
  content: string
  recipient: string
  status: 'PENDING' | 'SUCCESS' | 'FAILED'
  error_message?: string
  sent_at: string
}

// 告警规则API
export const alarmRuleApi = {
  // 获取规则列表
  getRules: (params?: {
    page?: number
    page_size?: number
    search?: string
    diagnosis_type?: DiagnosisType
    severity?: string
    is_enabled?: boolean
  }): Promise<ApiResponse<AlarmRule[]>> => {
    return request.get('/api/v1/alarm-rules/', { params })
  },

  // 获取规则详情
  getRule: (id: number): Promise<ApiResponse<AlarmRule>> => {
    return request.get(`/api/v1/alarm-rules/${id}`)
  },

  // 创建规则
  createRule: (data: AlarmRuleCreate): Promise<ApiResponse<AlarmRule>> => {
    return request.post('/api/v1/alarm-rules/', data)
  },

  // 更新规则
  updateRule: (id: number, data: AlarmRuleUpdate): Promise<ApiResponse<AlarmRule>> => {
    return request.put(`/api/v1/alarm-rules/${id}`, data)
  },

  // 删除规则
  deleteRule: (id: number): Promise<ApiResponse<void>> => {
    return request.delete(`/api/v1/alarm-rules/${id}`)
  },

  // 切换启用状态
  toggleRule: (id: number, is_enabled: boolean): Promise<ApiResponse<AlarmRule>> => {
    return request.put(`/api/v1/alarm-rules/${id}/toggle`, { is_enabled })
  },

  // 获取统计信息
  getStats: (): Promise<ApiResponse<{
    enabled_count: number
    critical_count: number
    triggered_today: number
    total_count: number
  }>> => {
    return request.get('/api/v1/alarm-rules/stats')
  }
}

// 通知渠道API
export const notificationChannelApi = {
  // 获取渠道列表
  getChannels: (params?: {
    page?: number
    page_size?: number
    search?: string
    type?: string
    is_enabled?: boolean
  }): Promise<ApiResponse<NotificationChannel[]>> => {
    return request.get('/api/v1/notification-channels/', { params })
  },

  // 获取渠道详情
  getChannel: (id: number): Promise<ApiResponse<NotificationChannel>> => {
    return request.get(`/api/v1/notification-channels/${id}`)
  },

  // 创建渠道
  createChannel: (data: NotificationChannelCreate): Promise<ApiResponse<NotificationChannel>> => {
    return request.post('/api/v1/notification-channels/', data)
  },

  // 更新渠道
  updateChannel: (id: number, data: NotificationChannelUpdate): Promise<ApiResponse<NotificationChannel>> => {
    return request.put(`/api/v1/notification-channels/${id}`, data)
  },

  // 删除渠道
  deleteChannel: (id: number): Promise<ApiResponse<void>> => {
    return request.delete(`/api/v1/notification-channels/${id}`)
  },

  // 测试通知
  testChannel: (id: number, data: {
    title: string
    content: string
    recipient?: string
  }): Promise<ApiResponse<void>> => {
    return request.post(`/api/v1/notification-channels/${id}/test`, data)
  },

  // 获取通知日志
  getLogs: (id: number, params?: {
    page?: number
    page_size?: number
    status?: string
  }): Promise<ApiResponse<NotificationLog[]>> => {
    return request.get(`/api/v1/notification-channels/${id}/logs`, { params })
  },

  // 获取统计信息
  getStats: (): Promise<ApiResponse<{
    enabled_count: number
    sent_today: number
    success_rate: number
    total_count: number
  }>> => {
    return request.get('/api/v1/notification-channels/stats')
  }
}

// 诊断统计API
export const diagnosisStatsApi = {
  // 获取统计概览
  getOverview: (days?: number): Promise<ApiResponse<DiagnosisStats>> => {
    return request.get('/api/v1/diagnosis/stats/overview', { params: { days } })
  }
}